pkg/tfpfbridge/schemashim/attr.go

```diff
@@ -28,3 +47,59 @@ type attr interface {
 	GetDescription() string
 	GetMarkdownDescription() string
 }
+
+// Dropping to reflection here since GetAttributes() refers to internal types from terraform-plugin-framework and cannot
+// be easily called safely.
+func getAttributes(a interface{}) map[string]attrLike {
+	attrs := map[string]attrLike{}
+	attrMap := callGetAttributes(reflect.ValueOf(a))
+
+	if attrMap.IsNil() {
+		return attrs
+	}
+
+	// Sometimes need to call a.GetAttributes().GetAttributes() to get to a map.
+	if attrMap.Type().Kind() != reflect.Map {
+		attrMap = callGetAttributes(attrMap)
+	}
+
+	if attrMap.Type().Kind() != reflect.Map {
```

Comment:
so if it's not a map after two calls, we panic? Do we catch this panic somewhere?
what is the condition under which we cannot translate this particular attribute? does it mean we're having bad input?
 Does this panic cause the entire provider to fail to build?

Reply:
Yes it would. I think I'd like that actually - this is essentially an assert that should not happen. Schema generation should not have these issues, if it does, it means some assumptions here are broken and we need to fix it. Fortunately this code does not run at runtime in the bridged provider at all so blast radius is only pertaining to builds.